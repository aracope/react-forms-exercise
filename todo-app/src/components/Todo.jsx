import React from "react";

function Todo({ id, task, remove }) {
  return (
    <div>
      <span>{task}</span>
      <button onClick={() => remove(id)}>X</button>
    </div>
  );
}

export default Todo;
