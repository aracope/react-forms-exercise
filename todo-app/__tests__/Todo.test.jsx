import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";   
import Todo from "../src/components/Todo";

it("renders without crashing", () => {
  render(<Todo id="1" task="Test" remove={() => { }} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Todo id="1" task="Test" remove={() => { }} />);
  expect(asFragment()).toMatchSnapshot();
});

it("displays task", () => {
  render(<Todo id="1" task="Test task" remove={() => { }} />);
  expect(screen.getByText("Test task")).toBeInTheDocument();
});

it("calls remove on click", async () => {
  const mockRemove = jest.fn();
  render(<Todo id="1" task="Test" remove={mockRemove} />);
  
  await userEvent.click(screen.getByText("X"));  
  expect(mockRemove).toHaveBeenCalledWith("1");
});
