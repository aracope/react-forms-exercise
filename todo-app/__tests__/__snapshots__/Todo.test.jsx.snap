// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <span>
      Test
    </span>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
